### PR TITLE
Switching from Java Logger to slf4j in EPH

### PR DESCRIPTION
**Pull request by sabeegrewal**
This just swaps all java logging with slf4j.

The java logging was centralized within EventProcessorHost.java. To stick to slf4j conventions, I 'de-centralized' the logging by creating a Logger within each .java file.

### CODE REVIEW

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/Pump.java [hunk 1]
```diff
@@ -83,17 +87,19 @@ public Void call() throws Exception
...
+
+    		TRACE_LOGGER.info(LoggingUtils.withHostAndPartition(this.host.getHostName(), partitionId, "removing pump"));
     		this.pumpStates.remove(partitionId);
     	}
     	else
     	{
     		// PartitionManager main loop tries to remove pump for every partition that the host does not own, just to be sure.
     		// Not finding a pump for a partition is normal and expected most of the time.
-    		this.host.logWithHostAndPartition(Level.FINER, partitionId, "no pump found to remove for partition " + partitionId);
+    		TRACE_LOGGER.info(LoggingUtils.withHostAndPartition(this.host.getHostName(), partitionId,
```
Comment (JamesBirdsall) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/Pump.java]: Suggest making this debug -- this is a spammy message that you're only going to care about when debugging in detail.

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionPump.java [hunk 2]
```diff
@@ -58,7 +61,7 @@ Void startPump()
             	// If the processor won't create or open, only thing we can do here is pass the buck.
             	// Null it out so we don't try to operate on it further.
             	this.processor = null;
-            	this.host.logWithHostAndPartition(Level.SEVERE, this.partitionContext, "Failed " + action, e);
+            	TRACE_LOGGER.warn(LoggingUtils.withHostAndPartition(this.host.getHostName(), this.partitionContext, "Failed " + action), e);
```
Comment (JamesBirdsall) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionPump.java]: Suggest error -- application writer will always want to log this. This is catastrophic, the sort of problem that should set off automated alerts if the user has them!

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java [hunk 3]
```diff
@@ -51,10 +54,11 @@
 				{
 					this.partitionIds = ehInfo.getPartitionIds();
 	
-					this.host.logWithHost(Level.FINE, "Eventhub " + this.host.getEventHubPath() + " count of partitions: " + ehInfo.getPartitionCount());
+					TRACE_LOGGER.info(LoggingUtils.withHost(this.host.getHostName(),
```
Comment (JamesBirdsall) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/PartitionManager.java]: This and the next one should be debug.

File: azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/LoggingUtils.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    static String withHost(String hostName, String logMessage) {
+        return "host " + hostName + ": " + logMessage;
+    }
+
+    static String withHostAndPartition(String hostName, String partitionId, String logMessage) {
+        return "host " + hostName + ": " + partitionId + ": " + logMessage;
+    }
+
+    static String withHostAndPartition(String hostName, PartitionContext context, String logMessage) {
+        return "host " + hostName + ": " + context.getPartitionId() + ": " + logMessage;
```
Comment (JamesBirdsall) [inline azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/LoggingUtils.java]: return LoggingUtils.withHostAndPartition(hostname, context.getPartitionId(), logMessage);

**Comment by sabeegrewal** [comment]
This handles #10

**Pull request merged**